compiler/rustc_mir_transform/src/coverage/mod.rs

```diff
@@ -136,20 +136,48 @@ fn create_mappings<'tcx>(
             .as_term()
     };
 
-    coverage_spans
-        .all_bcb_mappings()
-        .filter_map(|&BcbMapping { kind: bcb_mapping_kind, span }| {
+    let mut mappings = vec![];
+
+    mappings.extend(coverage_spans.all_bcb_mappings().filter_map(
+        |&BcbMapping { kind: bcb_mapping_kind, span }| {
             let kind = match bcb_mapping_kind {
                 BcbMappingKind::Code(bcb) => MappingKind::Code(term_for_bcb(bcb)),
-                BcbMappingKind::Branch { true_bcb, false_bcb } => MappingKind::Branch {
-                    true_term: term_for_bcb(true_bcb),
-                    false_term: term_for_bcb(false_bcb),
-                },
             };
             let code_region = make_code_region(source_map, file_name, span, body_span)?;
             Some(Mapping { kind, code_region })
-        })
-        .collect::<Vec<_>>()
+        },
+    ));
+
+    for arm_list in &coverage_spans.branch_arm_lists {
+        let mut arms_rev = arm_list.iter().rev();
+
+        let mut rest_counter = {
+            // The last arm's span is ignored.
+            let Some(&BcbBranchArm { span: _, bcb }) = arms_rev.next() else { continue };
+            coverage_counters.bcb_counter(bcb).expect("all relevant BCBs have counters")
+        };
+
+        for &BcbBranchArm { span, bcb } in arms_rev {
+            let true_counter =
+                coverage_counters.bcb_counter(bcb).expect("all relevant BCBs have counters");
+            let kind = MappingKind::Branch {
+                true_term: true_counter.as_term(),
+                false_term: rest_counter.as_term(),
+            };
+
+            mappings.extend::<Option<_>>(
+                try {
+                    let code_region = make_code_region(source_map, file_name, span, body_span)?;
+                    Mapping { kind, code_region }
+                },
+            );
```

Comment:
first time I see a `try` block in action.
But isn’t this the same as:
```suggestion
            if let Some(code_region) = make_code_region(source_map, file_name, span, body_span) {
                mappings.push(Mapping { kind, code_region });
            }
```

that would look a lot more readable.